examples/dmd/local_tw_csv.cpp

```diff
@@ -0,0 +1,640 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2022, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Compile with: make local_tw_csv
+//
+// =================================================================================
+//
+// Sample script for local DMD on heat conduction: heat_conduction_csv.sh
+//
+// Local serial DMD command:
+//   local_tw_csv -o local_csv_serial -rdim 16 -dtc 0.01
+//
+// Final-time prediction error (Last line in run/local_csv_serial/hc_par5_prediction_error.csv):
+//   0.000199922
+//
+// Local time windowing DMD command:
+//   local_tw_csv -o local_csv_tw -rdim 16 -nwinsamp 25 -dtc 0.01
+//
+// Final-time prediction error (Last line in run/local_csv_tw/hc_par5_prediction_error.csv):
+//   0.000334656
+//
+// =================================================================================
+//
+// Description: Local time windowing DMD on general CSV datasets.
+//
+// User specify file locations and names by -list LIST_DIR -train-set TRAIN_LIST -test-set TEST_LIST -data DATA_DIR -var VAR_NAME -o OUT_DIR
+//
+// File structure:
+// 1. LIST_DIR/TRAIN_LIST.csv             -- each row specifies one training DATASET
+// 2. LIST_DIR/TEST_LIST.csv              -- each row specifies one testing DATASET
+// 3. LIST_DIR/DATASET.csv                -- each row specifies one STATE in DATASET
+// 4. DATA_DIR/dim.csv                    -- specifies the dimension of VAR_NAME
+// 5. DATA_DIR/DATASET/tval.csv           -- specifies the time instances
+// 6. DATA_DIR/DATASET/STATE/VAR_NAME.csv -- each row specifies one value of VAR_NAME of STATE
```

Comment:
The solution field CSV files have one line for each unknown, written with 6 digits of accuracy. It seems that hdf5 output would be more accurate (more digits). The function `Vector::write` can be used.

Reply:
Another option is if you are using CSVDatabase to write out these files. You can instead increase the precision to 16 or some value equivalent to the HDF5Database. It is a one-liner.

https://stackoverflow.com/questions/5907031/printing-the-correct-number-of-decimal-points-with-cout

Reply:
Another option is if you are using CSVDatabase to write out these files. You can instead increase the precision to 16 or some value equivalent to the HDF5Database. It is a one-liner.

https://stackoverflow.com/questions/5907031/printing-the-correct-number-of-decimal-points-with-cout

Reply:
Set a higher precision of 10 for putDoubleArray and leave precision as a variable (default 6) in putDoubleVector and putComplexVector.